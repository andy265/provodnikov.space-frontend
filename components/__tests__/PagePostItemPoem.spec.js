const { createRenderer } = require('vue-server-renderer')
const { shallow } = require('vue-test-utils')
jest.mock('~/components/HasModelMixin', () => require('HasModelMixin'))
const PagePostItemPoem = require('~/components/PagePostItemPoem')

describe('PagePostItemPoem.vue', () => {
  it('has same HTML structure when state is "index"', () => {
    let Component = Object.assign({}, PagePostItemPoem, { components: {} })
    let wrapper = shallow(Component, {
      propsData: {
        state: 'index',
        isPreview: false,
        postUrl: 'http://localhost/post/post_url',
        value: {
          name: 'some name',
          text: 'line 1\nline 2\nline 3\nline 4'
        }
      }
    })
    let renderer = createRenderer()
    // eslint-disable-next-line
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })
})
