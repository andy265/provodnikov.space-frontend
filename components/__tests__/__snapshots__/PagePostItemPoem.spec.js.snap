// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagePostItemPoem.vue has same HTML structure when state is "index" 1`] = `
<div class="page-post-item-poem">
    <p class="page-post-item-poem__name">some name</p>
    <p class="page-post-item-poem__text">line 1 line 2 line 3 line 4</p>
</div>
`;
